contrib/cirrus/mac_cleanup.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+# This script is intended to be called by Cirrus-CI on a Mac M1 persistent workier.
+# It performs a best-effort attempt at cleaning up from one task execution to the next.
+# Since it run both before and after tasks, it must exit cleanly if there was a cleanup
+# failure (i.e. file or directory not found).
+
+set +ex
```

Comment:
I'd much rather `+e -x` for something that will need debugging one day

Reply:
Oops, my bad.  I always get confused about `-x` and `+x`.  You are correct, I was intending `-x` to enable debugging.  Good catch!

Reply:
Fix staged.